packages/dev/core/src/Misc/halton2DSequence.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Class for generating 2D Halton sequences.
+ * From https://observablehq.com/@jrus/halton
+ */
+export class Halton2DSequence {
+    private _curIndex = 0;
+    private _sequence: number[] = [];
+    private _numSamples = 0;
+    private _width: number;
+    private _height: number;
+    private _baseX;
+    private _baseY;
+
+    /**
+     * The x coordinate of the current sample.
+     */
+    public readonly x = 0;

```

Comment:
Should it be readonly if we update it often ?

Reply:
I made it `readonly` so that external code wouldn't try to set a value. And a getter seemed overkill, to get a simple value (I assumed a getter would be less efficient than a direct property).